File: golden_ticket_enterprise/lib/entities/main_tag.dart

import 'package:golden_ticket_enterprise/entities/subTag.dart';

class MainTag {
  final int tagID;
  final String tagName;
  final List<SubTag> subTags;

  MainTag({required this.tagID, required this.tagName, required this.subTags});

  factory MainTag.fromJson(Map<String, dynamic> json) {
    List<SubTag> subTags = [];

    for(var data in json['subTags']){
        subTags.add(SubTag.fromJson(data));
    }
    return MainTag(
        tagID: json['mainTagID'],
        tagName: json['mainTagName'] ?? "",
        subTags: subTags
      );
  }

  Map<String, dynamic> toJson() {
    return {
      'mainTagID': tagID,
      'mainTagName': tagName,
      'subTags': subTags,
    };
  }

}

File: GoldenTicket/GoldenTicket/Entities/MainTag.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
namespace GoldenTicket.Entities
{
    [Table("tblMainTag")]
    public class MainTag
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int TagID { get; set; }
        [Required]
        public string? TagName { get; set; }
        public ICollection<SubTag> ChildTags { get; set; } = new List<SubTag>();
    }

    public class MainTagDTO
    {
        public int MainTagID { get; set; }
        public string? MainTagName { get; set; }
        public List<SubTagDTO>? SubTags { get; set; }
    }

}
